src/redux/sagas/motions/createEditDomainMotion.ts

```diff
@@ -35,10 +33,175 @@ import {
+type CreateTransactionPayloadMotion = {
+  motionDomainId: number;
+};
+
+type CreateTransactionPayload = {
+  base: CreateTransactionPayloadBase;
+  motionPayload: CreateTransactionPayloadMotion;
+};
+
+function* getCreateTransactionPayload(payload: CreateTransactionPayload) {
+  const {
+    base: {
+      isMultiSig,
+      colonyAddress,
+      batchKey,
+      metaId,
+      index,
+      colonyDomains,
+      isCreateDomain,
+      domainId,
+      action,
+      colonyRoles,
+    },
+  } = payload;
+
+  const transactionParams = {
+    context: isMultiSig
+      ? ClientType.MultisigPermissionsClient
+      : ClientType.VotingReputationClient,
+    methodName: TRANSACTION_METHODS.CreateMotion,
+    identifier: colonyAddress,
+    group: {
+      key: batchKey,
+      id: metaId,
+      index,
+    },
+    params: [] as MethodParams,
+    ready: false,
+  };
+
+  const colonyManager = yield getColonyManager();
+
+  const colonyClient = yield colonyManager.getClient(
+    ClientType.ColonyClient,
+    colonyAddress,
+  );
+
+  // eslint-disable-next-line no-warning-comments
+  // TODO: Use Sam's upcoming list of required permissions as the source of truth
+  // src/constants/actions.ts: PERMISSIONS_NEEDED_FOR_ACTION
+  const requiredRoles = [ColonyRole.Architecture];
+
+  const userAddress = yield colonyClient.signer.getAddress();
+
+  if (isMultiSig) {
+    const [permissionDomainId, childSkillIndex] = yield call(
+      getPermissionProofsLocal,
+      {
+        networkClient: colonyClient.networkClient,
+        colonyRoles,
+        colonyDomains,
+        requiredDomainId: Id.RootDomain,
+        requiredColonyRole: requiredRoles,
+        permissionAddress: userAddress,
+        isMultiSig,
+      },
+    );
+
+    const encodedAction = colonyClient.interface.encodeFunctionData(
+      action,
+      [permissionDomainId, childSkillIndex, domainId, '.'], // domainMetadataIpfsHash
+    );
+
+    transactionParams.params = [
+      Id.RootDomain,
+      childSkillIndex,
+      [isCreateDomain ? colonyAddress : AddressZero],
+      [encodedAction],
+    ];
+  } else {
+    const rootDomain = colonyDomains.find((domain) =>
+      BigNumber.from(domain.nativeId).eq(Id.RootDomain),
+    );
+
+    if (!rootDomain) {
+      throw new Error('Cannot find rootDomain in colony domains');
+    }
+
+    const { motionDomainId } = payload.motionPayload;
+
+    const { skillId } = yield call(
+      [colonyClient, colonyClient.getDomain],
+      motionDomainId,
+    );
+
+    const {
+      key: reputationKey,
+      value: reputationValue,
+      branchMask: reputationBranchMask,
+      siblings: reputationSiblings,
+    } = yield call(colonyClient.getReputation, skillId, AddressZero);
+
+    const motionChildSkillIndex = yield call(getChildIndexLocal, {
+      networkClient: colonyClient.networkClient,
+      parentDomainNativeId: rootDomain.nativeId,
+      parentDomainSkillId: rootDomain.nativeSkillId,
+      domainNativeId: rootDomain.nativeId,
```

Comment:
should we call this with the rootDomain or should we call it with the motionDomain here? So we most likely would need a lookup for this. Since we reuse a single saga for editing and creating, this is a bit confusing.

Reply:
I will need some extra clarification for this part @bassgeta. Can we pair up about this over a call?